### PR TITLE
fix: not to call sync with SHA of parent event if it is pr event

### PR DESCRIPTION
**Pull request by sakka2**
## Context
Now, pipeline `sync()` method is called with parent event sha even if it is PR event.
This cause to create a new job with parent PR event config as a commit event.
Created new job doesn't have `PR-` prefix.

## Objective
Do not call `sync()` with parent event sha if it is PR event.

### CODE REVIEW

File: lib/eventFactory.js [hunk 1]
```diff
@@ -421,7 +421,7 @@ class EventFactory extends BaseFactory {
             // Sync pipeline to make sure workflowGraph is generated
             .then((p) => {
                 // Sync pipeline with the parentEvent sha and create jobs based on that sha
-                if (parentEventId) {
+                if (parentEventId && !prRef) {
```
Comment (tk3fftk) [inline lib/eventFactory.js:424]: It's ok to check the event is PR or not. In my opinion, the intention looks unclear.

**Review by tk3fftk — APPROVED** [review]

**Review by wdstar — APPROVED** [review]

**Review by minzcmu — APPROVED** [review]

**Comment by d2lam** [comment]
Snyk failed. Closing to retrigger.

File: lib/eventFactory.js (hunk 1 above)
Comment (jithine) [reply lib/eventFactory.js:424]: I am not clear on PR title vs code change, shouldn't this change be on line [428](https://github.com/screwdriver-cd/models/pull/350/files#diff-26258d311a2f870653c61d02805df225R428) below ?

Comment (sakka2) [reply lib/eventFactory.js:424]: What I wanted to say is pipeline sync method with the parent event sha should not be called when PR event.
I will change the title.

**Pull request merged**